src/BondingERC20TokenFactory.sol

```diff
@@ -61,6 +70,19 @@ contract BondingERC20TokenFactory is IBondingERC20TokenFactory, Ownable {
     return address(_bondingERC20Token);
   }
 
+  function claimFees() public onlyOwner {
+    emit FeesSent(address(this).balance);
+    (bool success, ) = treasury.call{value: address(this).balance}("");
+    if (!success) {
+        revert FeeSendingFailed();
+    }
+  }
+
+  function updateCreationFee(uint256 newCreationFee) public onlyOwner {
+    emit CreationFeeUpdated(newCreationFee, creationFee);
```

Comment:
So the new consensus is to emit event before the operation?

Reply:
it's not the rule but I did it for gas savings. `creationFee` is state variable and if we update it before emitting the event, we will need to cache that variable to get the old value because it was already updated. This doesn't cause any security issues because no event will be emitted if transaction reverts.